### DISCUSSION TITLE
Speed up pages with many linked references

### DISCUSSION THREAD

**Discussion by agentydragon** [athensresearch/athens#890] [Archive]
I'm playing with giving my database some structure. I currently put a `[[Hierarchy]]` block on all pages, with some parallel hierarchies for categorizing things. For example

```
Add search to [[Athens Research]]
===

* [[Hierarchy]]:
  * [[Task]] (it's a task)
  * [[Athens Research]] (within Athens Research)
```

Unfortunately having this block in every page makes opening the `Hierarchy` page quite slow. I suspect it might be because loading all the linked references takes time, and might not be super efficient. Would be nice if we could paginate them, or optimize the case.

**Comment by jefftangx** [comment]
Unlinked References are quite slow as well, which is why they are closed by default. It might make sense to just open the page by default, and then progressively query and load linked and unlinked. A "loading" UI could be displayed, and a little indicator showing when the query is done.
